udp/udp_all.go

```diff
@@ -64,6 +64,22 @@ func (ua *Addr) Copy() *Addr {
 	return &nu
 }
 
+type AddrSlice []*Addr
+
+func (a AddrSlice) Equal(b AddrSlice) bool {
```

Comment:
This could ostensibly be implemented using [`slices.EqualFunc`](https://pkg.go.dev/golang.org/x/exp/slices#EqualFunc).